File: home-control/src/main/kotlin/space/termite/homecontrol/rules/TriggeredAction.kt
package space.termite.homecontrol.rules

/**
 * A concrete action to execute as a consequence of a rule match: publish [payload] to
 * [destinationTopic] on the MQTT broker.
 */
data class TriggeredAction(
    val destinationTopic: String,
    val payload: String,
    val retain: Boolean = false,
)

File: home-control-api/src/rules/triggered-action.ts
/**
 * An action to execute after a rule match: publish [payload] to [destinationTopic].
 *
 * The rule that matched the origin topic chooses [destinationTopic]; a single event
 * may fan out to several actions on different topics (e.g. `zigbee2mqtt/switch` in,
 * `zigbee2mqtt/plug` out).
 */
export interface TriggeredAction {
  destinationTopic: string;
  payload: string;
  retain?: boolean;
}
